diesel_derives/src/lib.rs

```diff
@@ -597,6 +597,151 @@ pub fn derive_sql_type(input: TokenStream) -> TokenStream {
 /// #     Ok(())
 /// # }
 /// ```
+///
+/// ## Custom Aggregate Functions
+/// Custom aggregate functions can be created in SQLite by adding an `#[aggregate]`
+/// attribute inside of `sql_function`. `register_impl` needs to be called on the generated function with
+/// a type implementing the [SqliteAggregateFunction](../diesel/sqlite/trait.SqliteAggregateFunction.html) trait
+/// as a type parameter as shown in the examples below.
+///
+/// ```rust
+/// # extern crate diesel;
+/// # use diesel::*;
+/// #
+/// # #[cfg(feature = "sqlite")]
+/// # fn main() {
+/// #   run().unwrap();
+/// # }
+/// #
+/// # #[cfg(not(feature = "sqlite"))]
+/// # fn main() {
+/// # }
+/// use diesel::sql_types::Integer;
+/// # #[cfg(feature = "sqlite")]
+/// use diesel::sqlite::SqliteAggregateFunction;
+/// use std::ops::AddAssign;
+///
+/// sql_function! {
+///     #[aggregate]
+///     fn my_sum(x: Integer) -> Integer;
+/// }
+///
+/// #[derive(Default)]
+/// struct MySum<T> { sum: T }
+///
+/// # #[cfg(feature = "sqlite")]
+/// impl<T: Default + Copy + AddAssign> SqliteAggregateFunction<T> for MySum<T> {
```

Comment:
Probably a small thing but I think I would prefer having at least one example without a generic implementation of `SqliteAggregateFunction` to show it's totally fine to not use generics here.
(In the end that's the example people will copy and modify as first step. The simple at least one example, the better I would say)